orquesta/specs/native/v1/models.py

```diff
@@ -139,9 +153,63 @@ def __init__(self, *args, **kwargs):
             self.action = action_spec[:action_spec.index(' ')]
             self.input = input_spec
 
+    def has_items(self):
+        return hasattr(self, 'with') and getattr(self, 'with', None) is not None
+
+    def get_items_spec(self):
+        return getattr(self, 'with', None)
+
     def has_join(self):
         return hasattr(self, 'join') and self.join
 
+    def render(self, in_ctx):
+        action_specs = []
+
+        if not self.has_items():
+            action_spec = {
+                'action': expr.evaluate(self.action, in_ctx),
+                'input': expr.evaluate(getattr(self, 'input', {}), in_ctx)
+            }
+
+            action_specs.append(action_spec)
+        else:
+            items_spec = self.get_items_spec()
+
+            items_expr = (
+                items_spec.items.strip() if 'in' not in items_spec.items
+                else items_spec.items[items_spec.items.index('in') + 2:].strip()
+            )
+
+            items = expr.evaluate(items_expr, in_ctx)
+
+            if not isinstance(items, list):
+                raise TypeError('The value of "%s" is not type of list.' % items_expr)
+
+            item_keys = (
+                None if 'in' not in items_spec.items
+                else items_spec.items[:items_spec.items.index('in')].replace(' ', '').split(',')
+            )
+
+            for i in range(0, len(items)):
```

Comment:
Could simplify to enumerate. Not a blocker.

Reply:
Thanks. I will refactor.